.github/workflows/building.yml

```diff
@@ -0,0 +1,37 @@
+name: Building
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  build-check:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        project: ['upstream-tag-on-merge', 'upstream-tag-sync']
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Bun
+        uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: latest
+
+      - name: Install Dependencies
+        run: bun install
+        working-directory: ./${{ matrix.project }}
+
+      - name: Build
+        run: bun run build
+        working-directory: ./${{ matrix.project }}
+
+      - name: Check if build is committed
+        run: |
+          if [ -n "$(git status --porcelain)" ]; then
+            echo "Error: Build artifacts don't match. Please run 'bun run build' and commit the changes."
+            git status
+            exit 1
+          fi
```

Comment:
If I understand correctly this assumes that `bun` yields reproducible builds. I'm not familiar enough with it to know if it's a reasonable assumption (i.e. no false positives), but I guess we will find out later once it's merged.

Reply:
@devnoname120 I tested it in another private repo and it seems to work fine.